### PR TITLE
.ConfigureAwait(false) in SignalR client libraries

### LINKED ISSUE #39626: SignalR C# Client StartAsync() CancellationToken not respected
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Similar to https://github.com/dotnet/aspnetcore/issues/10059. I have a Client application (WPF .NET6) that uses the DI-framework which hosts a service on the background that connects to a SignalR hub. The cancellationtoken seems to not be respected and never finishes my task and thus blocks my application from exiting.

### Expected Behavior

I would expect that I receive a TaskCanceledException.

### Steps To Reproduce

To reproduce this you should have your SignalR server turned **off** (I don't think you even need a server!). In the below example the "await" start-task never finishes.

Example client code:
```
    internal class ServerProxy : IHostedService
    {
        private readonly HubConnection _connection;
        private readonly CancellationTokenSource _cts = new();
        private Task _startTask;

        public ServerProxy(IConfiguration configuration)
        {
            _connection = new HubConnectionBuilder()
                .WithUrl($"{configuration.GetValue<string>("ServerBaseAddress")}/messages")
                .WithAutomaticReconnect()
                .Build();
        }

        public Task StartAsync(CancellationToken cancellationToken)
        {
            _startTask = _connection.StartAsync(_cts.Token);
            return Task.CompletedTask;
        }

        public async Task StopAsync(CancellationToken cancellationToken)
        {
            if (_startTask != null && !_startTask.IsCompleted)
            {
                _cts.Cancel();
                try
                {
                    await _startTask;
                }
                catch (TaskCanceledException)
                {
                    // Do nothing
                }
            }

            if (_connection.State == HubConnectionState.Connected)
            {
                await _connection.StopAsync();
                await _connection.DisposeAsync();
            }
        }
    }
```

### Exceptions (if any)

n/a

### .NET Version

6.0.100

### Anything else?

Using the Package "Microsoft.AspNetCore.SignalR.Client" Version="6.0.1".

### PR DESCRIPTION
**Pull request by halter73**
# `.ConfigureAwait(false)` in SignalR client libraries

- Enable `CA2007` (Consider calling ConfigureAwait on the awaited task) for SignalR client libraries and shared libraries
- Don't use `ForceAsync()` in ignalR client libraries and shared libraries
  - The previous use of `ForceAsync()` could still cause deadlocks of the code using the client tried to block a UI thread (or any one-at-a-time `SynchronizationContext`) because internal SignalR logic would try to post back to it.

Fixes #39626
Partially addresses #31132

### CODE REVIEW

File: AspNetCore.sln [hunk 1]
```diff
@@ -8804,22 +8804,6 @@ Global
 		{E2461809-D2EA-436D-B5C3-8A9EE0A283B8}.Release|x64.Build.0 = Release|Any CPU
 		{E2461809-D2EA-436D-B5C3-8A9EE0A283B8}.Release|x86.ActiveCfg = Release|Any CPU
 		{E2461809-D2EA-436D-B5C3-8A9EE0A283B8}.Release|x86.Build.0 = Release|Any CPU
-		{CAFD1885-B87B-4A7A-8BE6-86B0C238C2B1}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (pranavkm) [inline AspNetCore.sln:8807]: Bad sln?

**Review by pranavkm — APPROVED** [review]
👍🏽 We should also consider enabling this for things that aren't exclusively ASP.NET Core (e.g. DataProtection, Extensions.Features, JSInterop etc).

**Comment by davidfowl** [comment]
Why are we using ConfigureAwait on the server side?

File: src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs [hunk 2]
```diff
@@ -709,7 +709,7 @@ private static void CloneHttpContext(HttpContext context, HttpConnectionContext
             // No connection with that ID: Not Found
             context.Response.StatusCode = StatusCodes.Status404NotFound;
             context.Response.ContentType = "text/plain";
-            await context.Response.WriteAsync("No Connection with that ID");
+            await context.Response.WriteAsync("No Connection with that ID").ConfigureAwait(false);
```
Comment (davidfowl) [inline src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:712]: Why?

File: src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs [hunk 3]
```diff
@@ -732,7 +732,7 @@ private static void CloneHttpContext(HttpContext context, HttpConnectionContext
         {
             // No connection with that ID: Not Found
             context.Response.StatusCode = StatusCodes.Status404NotFound;
-            await context.Response.WriteAsync("No Connection with that ID");
+            await context.Response.WriteAsync("No Connection with that ID").ConfigureAwait(false);
```
Comment (davidfowl) [inline src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:735]: Why?

File: src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs [hunk 4]
```diff
@@ -480,7 +480,7 @@ private async Task ProcessSend(HttpContext context, HttpConnectionDispatcherOpti
 
     private async Task ProcessDeleteAsync(HttpContext context)
```
Comment (davidfowl) [inline src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs:481]: Why?

File: AspNetCore.sln (hunk 1 above)
Comment (halter73) [reply AspNetCore.sln:8807]: Yes.

**Comment by halter73** [comment]
> Why are we using ConfigureAwait on the server side?

Fixed.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**